mod/beacon/validator/propose.go

```diff
@@ -102,46 +83,16 @@ func (s *Service[
 		return blk, sidecars, ErrNilPayload
 	}
 
-	// If we get returned a nil blobs bundle, we should return an error.
-	blobsBundle := envelope.GetBlobsBundle()
-	if blobsBundle == nil {
-		return blk, sidecars, ErrNilBlobsBundle
-	}
-
-	// Set the KZG commitments on the block body.
-	body.SetBlobKzgCommitments(blobsBundle.GetCommitments())
-
-	depositIndex, err := st.GetEth1DepositIndex()
-	if err != nil {
-		return blk, sidecars, ErrNilDepositIndexStart
-	}
-
-	// Dequeue deposits from the state.
-	deposits, err := s.bsb.DepositStore(ctx).GetDepositsByIndex(
-		depositIndex,
-		s.chainSpec.MaxDepositsPerBlock(),
+	// Create a new empty block from the current state.
+	blk, err = s.getEmptyBeaconBlockForSlot(
+		st, requestedSlot,
 	)
 	if err != nil {
 		return blk, sidecars, err
 	}
 
-	// Set the deposits on the block body.
-	body.SetDeposits(deposits)
-
-	// Set the KZG commitments on the block body.
-	body.SetBlobKzgCommitments(blobsBundle.GetCommitments())
-
-	// TODO: assemble real eth1data.
-	body.SetEth1Data(&types.Eth1Data{
-		DepositRoot:  primitives.Bytes32{},
-		DepositCount: 0,
-		BlockHash:    common.ZeroHash,
-	})
-
-	// Set the execution data.
-	if err = body.SetExecutionData(
-		envelope.GetExecutionPayload(),
-	); err != nil {
+	sidecars, err = s.buildBlockBody(ctx, st, blk, reveal, envelope)
```

Comment:
wait actually, why does this function need to return the sidecars?

Reply:
hmm seems like it does not have to returns the sidecar. My mistake

Reply:
updated sir